### PR TITLE
fix: add timeout for RPC calls to Secondary SP

### PR DESCRIPTION
**Pull request by alexgao001**
### Description

Set a timeout for calls to secondary SP, currently it is not config and the response from secondary SP might take over 2 min. This would block the executor to handle tasks. And if the Manager server can not receive a response from the executor, task will be dispatch which would cause duplicate task. 

### Rationale

tell us why we need these changes...

### Example

add an example CLI or API response...

### Changes

Notable changes: 
* add each change in a bullet point here
* ...

### CODE REVIEW

File: modular/executor/execute_replicate.go [hunk 1]
```diff
@@ -302,7 +313,9 @@ func (e *ExecuteModular) doneReplicatePiece(ctx context.Context, rTask coretask.
 	receive.SetSignature(taskSignature)
 	doneReplicateTime := time.Now()
 	if err = retry.Do(func() error {
```
Comment (will-2012) [inline modular/executor/execute_replicate.go:315]: Why is there no callback parameter of retry.OnRetry?

**Review by will-2012 — APPROVED** [review]

File: modular/executor/execute_replicate.go [hunk 2]
```diff
@@ -250,11 +253,19 @@ func (e *ExecuteModular) doReplicatePiece(ctx context.Context, waitGroup *sync.W
...
-			"redundancy_idx", redundancyIdx, "error", err)
+		// timeout for single piece replication
+		ctxWithTimeout, cancel := context.WithTimeout(ctx, replicateTimeOut)
+		defer cancel()
+		return e.baseApp.GfSpClient().ReplicatePieceToSecondary(ctxWithTimeout, spEndpoint, receive, data)
+	}, RtyAttem,
+		RtyDelay,
+		RtyErr,
+		retry.OnRetry(func(n uint, err error) {
+			log.CtxErrorw(ctx, "failed to replicate piece", "segment_idx", segmentIdx, "redundancy_idx", redundancyIdx, "error", err, "attempt", n, "max_attempts", RtyAttNum)
```
Comment (will-2012) [inline modular/executor/execute_replicate.go:266]: It is best to print endpont and error in the context.

File: modular/executor/execute_replicate.go [hunk 3]
```diff
@@ -250,11 +253,19 @@ func (e *ExecuteModular) doReplicatePiece(ctx context.Context, waitGroup *sync.W
...
+		})); err != nil {
+		log.CtxErrorw(ctx, "failed to replicate piece", "segment_idx", segmentIdx, "redundancy_idx", redundancyIdx, "error", err)
```
Comment (will-2012) [inline modular/executor/execute_replicate.go:268]: Ditto, add the `spEndpoint`.

**Review by zjubfd — APPROVED** [review]

**Review by zjubfd — APPROVED** [review]

File: modular/executor/execute_replicate.go (hunk 1 above)
Comment (alexgao001) [reply modular/executor/execute_replicate.go:315]: added

File: modular/executor/execute_replicate.go (hunk 2 above)
Comment (alexgao001) [reply modular/executor/execute_replicate.go]: added

File: modular/executor/execute_replicate.go (hunk 3 above)
Comment (alexgao001) [reply modular/executor/execute_replicate.go]: added

**Pull request merged**